### PR TITLE
fix regression in trackingspamprevention and warning in bound parameter checking code

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Changes:

* allow option to disable allowed host check since trackingspamprevention makes a request to amazonaws
* bound parameters can be a single string rather than an array
* disable testdox and add build message output to enable it if the build hangs

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: core/bootstrap.php [hunk 1]
```diff
@@ -37,6 +37,7 @@
 @date_default_timezone_set('UTC');
 
 disableEaccelerator();
+printTestDoxHint();
```
Comment (sgiehl) [inline core/bootstrap.php]: Seems that currently printed in any case. Shouldn't that only be shown when running tests and if testdox isn't enabled?

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: core/bootstrap.php (hunk 1 above)
Comment (diosmosis) [reply core/bootstrap.php]: @sgiehl I can add this, but given testdox will rarely be used, and will only be set manually, I didn't think it was worth the added complexity. Let me know if this is still an important change.

**Comment by diosmosis** [comment]
Also note: after merging, TrackingSpamPrevention's minimum required matomo will need to be updated

**Comment by sgiehl** [comment]
The problem is that most tests are failing. Guess it might make more sense
to move that to the console command that runs the tests maybe

**Comment by diosmosis** [comment]
@sgiehl I see, nevermind, phpstorm gave me the wrong bootstrap file. It should be in the test's bootstrap.php.

**Pull request merged**